sdk/Javascript/src/Platforms/Platform.FxOS.js

```diff
@@ -0,0 +1,277 @@
+// ----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ----------------------------------------------------------------------------
+
+/// <reference path="..\Generated\MobileServices.DevIntellisense.js" />
+/*global $__fileVersion__:false, $__version__:false */
+
+var _ = require('Extensions');
+var Validate = require('Validate');
+var Promises = require('Promises');
```

Comment:
Does FireFox OS have a built in Promise object like WInJS we could look at using instead of this simple A+ implementation?